### PR TITLE
Blog: Postgres Cafe Episode 1 - pgroll

### PR DESCRIPTION
**Pull request by CezzaineZ**
## Summary
https://docs.google.com/document/d/1QrA2XzU1iMaPcFgT0zmPG-Kb_Qi_FamO0THRpbmu1Vw/edit?tab=t.0

### Blog Release Checklist

- \[ ] Release on: \<mm/dd/yy>
- \[ ] Published on dev.to (connected to Xata org)
- \[ ] OG image included
- \[ ] Social media is scheduled for this post

### CODE REVIEW

File: postgres-cafe-pgroll.mdx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+- Offering **schema version views** so applications can connect to the specific schema version they need, enabling zero-downtime migrations.
+- Supporting the ability to **pause in the “expand” phase**, giving teams flexibility to test changes before finalizing.
+
+pgroll provides a practical solution to the most pressing migration challenges, reducing downtime and making complex changes more manageable for developers.
+
+## Watch the full episode
+
+For an in-depth exploration of pgroll and its capabilities, watch the full episode here:
+
+<ArticleVideo platform="youtube" src="https://www.youtube.com/embed/WwaJd2c9whM?si=JQOz2YTr-VH6ZudG" />
```
Comment (CezzaineZ) [inline postgres-cafe-pgroll.mdx:45]:
```suggestion
<ArticleVideo platform="youtube" src="https://www.youtube.com/embed/1j07dnd1nSY?si=LBPoDKT98RAn7KFs" />
```

**Review by alexfrancoeur — APPROVED** [review]
LGTM

File: postgres-cafe-pgroll.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## pgroll’s approach: automating expand-contract
+
+The [expand-contract pattern](https://xata.io/blog/pgroll-expand-contract) is a safe and reliable approach for schema migrations. pgroll automates this multi-step process by:
+
+- Using **triggers to enable dual writes**, allowing applications to continue working with both old and new schema versions.
+- Offering **schema version views** so applications can connect to the specific schema version they need, enabling zero-downtime migrations.
+- Supporting the ability to **pause in the “expand” phase**, giving teams flexibility to test changes before finalizing.
+
+pgroll provides a practical solution to the most pressing migration challenges, reducing downtime and making complex changes more manageable for developers.
+
```
Comment (gulcin) [inline postgres-cafe-pgroll.mdx:40]: "pgroll provides a practical solution to the most pressing schema change challenges by enabling safe rollbacks, adjustable backfilling, and thorough testing, giving developer and DBA teams the confidence they need to implement database changes without downtime."

**Review by gulcin — CHANGES_REQUESTED** [review]

**Comment by gulcin** [comment]
A great write-up! Thanks @CezzaineZ , suggested a few minor changes.

File: postgres-cafe-pgroll.mdx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+authorEmail: gulcin@xata.io
+date: 12-18-2024
+tags: ['open-source', 'postgres', 'fpSchemaMigrations', 'fpPgroll', 'oss']
+published: true
+slug: postgres-cafe-pgroll
+ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/pgroll-postgres-cafe-og@2x.png
+---
+
+We’re excited to kick off [Postgres Cafe](https://www.youtube.com/watch?v=WwaJd2c9whM), a blog and video series from Xata in collaboration with [Data Bene](https://www.data-bene.io/en/), where we dive into innovative open-source tools for Postgres. Each episode highlights a new tool designed to make working with Postgres databases easier and more efficient.
+In this first episode, we explore [pgroll](https://pgroll.com/), our tool for achieving zero-downtime database schema changes, and the solutions it provides to common database migration challenges.
```
Comment (CezzaineZ) [inline postgres-cafe-pgroll.mdx:17]:
```suggestion
In this first episode, we explore [pgroll](https://pgroll.com/), our tool for achieving zero-downtime database schema changes, and the solutions it offers for overcoming common challenges in modifying database schemas.
```

File: postgres-cafe-pgroll.mdx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+# Episode 1: pgroll
+
+Schema changes are often challenging, especially for production databases where downtime is not an option. In this episode, we discuss pgroll, a tool designed to tackle these migration pain points with an approach that prioritizes reliability and uptime. Here’s a look at what this episode covered:
```
Comment (CezzaineZ) [inline postgres-cafe-pgroll.mdx:21]:
```suggestion
Schema changes are often challenging, especially for production databases where downtime is not an option. In this episode, we discuss pgroll, a tool designed to tackle these challenges with an approach that prioritizes reliability and uptime. Here’s a look at what this episode covered:
```

File: postgres-cafe-pgroll.mdx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+## Key challenges in schema changes
+
+Schema changes can lead to various issues, particularly around:
+
+- **Schema debt:** Development teams often accrue “schema debt” over time, adding rather than modifying fields to avoid risky changes. This buildup can lead to cluttered schemas that impact performance and maintainability.
+- **Exclusive locks:** Many schema changes in Postgres require exclusive locks, which can block reads and writes, risking significant downtime. pgroll’s approach helps minimize these locks, keeping applications responsive during migrations.
+- **Rollback complexity:** Rollbacks are often tricky and rarely tested, especially when schema changes can’t be easily undone. pgroll addresses this with a migration pattern that simplifies reverting changes without application downtime.
```
Comment (CezzaineZ) [inline postgres-cafe-pgroll.mdx:29]:
```suggestion
- **Rollback complexity:** Rollbacks are often tricky and rarely tested, especially when schema changes can’t be easily undone. pgroll addresses this with a pattern that simplifies reverting changes without application downtime.
```

File: postgres-cafe-pgroll.mdx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+## pgroll’s approach: automating expand-contract
+
+The [expand-contract pattern](https://xata.io/blog/pgroll-expand-contract) is a safe and reliable approach for schema migrations. pgroll automates this multi-step process by:
+
+- Using **triggers to enable dual writes**, allowing applications to continue working with both old and new schema versions.
+- Offering **schema version views** so applications can connect to the specific schema version they need, enabling zero-downtime migrations.
```
Comment (CezzaineZ) [inline postgres-cafe-pgroll.mdx:36]:
```suggestion
- Offering **schema version views** so applications can connect to the specific schema version they need, enabling zero-downtime alterations.
```

File: postgres-cafe-pgroll.mdx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+- Supporting the ability to **pause in the “expand” phase**, giving teams flexibility to test changes before finalizing.
+
+pgroll provides a practical solution to the most pressing migration challenges, reducing downtime and making complex changes more manageable for developers.
```
Comment (CezzaineZ) [inline postgres-cafe-pgroll.mdx:39]:
```suggestion
pgroll provides a practical solution to the most pressing schema change challenges, by enabling safe rollbacks, adjustable backfilling, and thorough testing, giving developer and DBA teams the confidence they need to implement database changes without downtime.
```

File: postgres-cafe-pgroll.mdx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+## Watch the full episode
+
+For an in-depth exploration of pgroll and its capabilities, watch the full episode here:
+
+<ArticleVideo platform="youtube" src="https://www.youtube.com/embed/1j07dnd1nSY?si=LBPoDKT98RAn7KFs" />
+
+## Stay tuned for more Postgres tools
+
+This is just the start of Postgres Cafe! [Subscribe to the playlist](https://www.youtube.com/playlist?list=PLf7KS0svgDP_zJmby3RMzzOVO45qLbruA) for future episodes that will feature more open-source tools like [pgstream](https://xata.io/pgstream) for managing real-time data streaming, [pg_benchmark](https://git.data-bene.io/PostgreSQL/pg_benchmark/) for executing PostgreSQL benchmarks, and more. Watch this space to learn how each tool can make working with Postgres smoother and more efficient.
```
Comment (CezzaineZ) [inline postgres-cafe-pgroll.mdx:49]:
```suggestion
This is just the start of Postgres Cafe! [Subscribe to the playlist](https://www.youtube.com/playlist?list=PLf7KS0svgDP_zJmby3RMzzOVO45qLbruA) for future episodes that will feature more open-source tools like [pgstream](https://xata.io/pgstream) for managing real-time data streaming, [StatsMgr](https://git.data-bene.io/cedric/statsmgr) for managing statistics in PostgreSQL, and more. Watch this space to learn how each tool can make working with Postgres smoother and more efficient.
```

File: postgres-cafe-pgroll.mdx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+Schema changes can lead to various issues, particularly around:
+
+- **Schema debt:** Development teams often accrue “schema debt” over time, adding rather than modifying fields to avoid risky changes. This buildup can lead to cluttered schemas that impact performance and maintainability.
+- **Exclusive locks:** Many schema changes in Postgres require exclusive locks, which can block reads and writes, risking significant downtime. pgroll’s approach helps minimize these locks, keeping applications responsive during migrations.
+- **Rollback complexity:** Rollbacks are often tricky and rarely tested, especially when schema changes can’t be easily undone. pgroll addresses this with a pattern that simplifies reverting changes without application downtime.
+
+## pgroll’s approach: automating expand-contract
+
+The [expand-contract pattern](https://xata.io/blog/pgroll-expand-contract) is a safe and reliable approach for schema migrations. pgroll automates this multi-step process by:
```
Comment (CezzaineZ) [inline postgres-cafe-pgroll.mdx:33]:
```suggestion
The [expand-contract pattern](https://xata.io/blog/pgroll-expand-contract) is a safe and reliable approach for schema changes. pgroll automates this multi-step process by:
```

**Review by gulcin — APPROVED** [review]

**Pull request merged**